src/components/ConnectToNetSuiteFlow/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect, useState} from 'react';
+import * as Expensicons from '@components/Icon/Expensicons';
+import PopoverMenu from '@components/PopoverMenu';
+import useLocalize from '@hooks/useLocalize';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import {getAdminPoliciesConnectedToNetSuite} from '@libs/actions/Policy/Policy';
+import Navigation from '@libs/Navigation/Navigation';
+import {useAccountingContext} from '@pages/workspace/accounting/AccountingContext';
+import type {AnchorPosition} from '@styles/index';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {ConnectToNetSuiteFlowProps} from './types';
+
+function ConnectToNetSuiteFlow({policyID}: ConnectToNetSuiteFlowProps) {
+    const {translate} = useLocalize();
+
+    const hasPoliciesConnectedToNetSuite = !!getAdminPoliciesConnectedToNetSuite()?.length;
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const [isReuseConnectionsPopoverOpen, setIsReuseConnectionsPopoverOpen] = useState(false);
+    const [reuseConnectionPopoverPosition, setReuseConnectionPopoverPosition] = useState<AnchorPosition>({horizontal: 0, vertical: 0});
+    const {integrationRefs} = useAccountingContext();
+
+    const threeDotsMenuContainerRef = integrationRefs?.current?.[CONST.POLICY.CONNECTIONS.NAME.NETSUITE];
+
+    const connectionOptions = [
+        {
+            icon: Expensicons.LinkCopy,
+            text: translate('workspace.common.createNewConnection'),
+            onSelected: () => {
+                Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_TOKEN_INPUT.getRoute(policyID));
+                setIsReuseConnectionsPopoverOpen(false);
+            },
+        },
+        {
+            icon: Expensicons.Copy,
+            text: translate('workspace.common.reuseExistingConnection'),
+            onSelected: () => {
+                Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_EXISTING_CONNECTIONS.getRoute(policyID));
+                setIsReuseConnectionsPopoverOpen(false);
+            },
+        },
+    ];
+
+    useEffect(() => {
+        if (!hasPoliciesConnectedToNetSuite) {
+            Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_TOKEN_INPUT.getRoute(policyID));
+            return;
+        }
+        setIsReuseConnectionsPopoverOpen(true);
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (threeDotsMenuContainerRef) {
+        if (!isSmallScreenWidth) {
```

Comment:
We can do that in one `if`
```suggestion
    if (threeDotsMenuContainerRef && !isSmallScreenWidth) {
```